lib/rex/ui/text/shell.rb

```diff
@@ -157,8 +157,8 @@ def run(&block)
           ret = run_single(line)
           # don't bother saving lines that couldn't be found as a
           # command, create the file if it doesn't exist, don't save dupes
-          if ret && self.histfile && line != @last_line
-            File.open(self.histfile, "a+") { |f| f.puts(line) }
+          if ret && self.histfile && line&.strip != @last_line&.strip
```

Comment:
The safe navigation operator here on `line` seems like a bad idea. If `line` is nil, and `@last_line` is not, then this test will evaluate to true because nil is not a string.

```suggestion
          if ret && self.histfile && line && line.strip != @last_line&.strip
```

Reply:
Agreed good point I didn't notice that oversight, thanks for pointing it out!

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/15085/commits/db8cb32776918e6e52ccc8cd594074c9d829d078